databricks/koalas/series.py

```diff
@@ -4795,11 +4795,11 @@ def dot(self, other) -> Union[Scalar, "Series"]:
         14
         """
         if isinstance(other, DataFrame):
-            raise ValueError(
-                "Series.dot() is currently not supported with DataFrame since "
-                "it will cause expansive calculation as many as the number "
-                "of columns of DataFrame"
-            )
+            col_results = []
+            with option_context("compute.ops_on_diff_frames", True):
+                for k, v in other._ksers.items():
+                    col_results.append((self * v).sum())
+            return Series(col_results, index=other.columns)
```

Comment:
This hits the same issues as https://github.com/databricks/koalas/pull/1136#pullrequestreview-333211768

Reply:
Modified!